src/flows/pipes/Visualization/ExportDashboardOverlay/Visualization.tsx

```diff
@@ -19,20 +26,23 @@ const Visualization: FC = () => {
   const [loading, setLoading] = useState(RemoteDataState.NotStarted)
   const {data} = useContext(PopupContext)
   const {query} = useContext(QueryContext)
+  const queryText = useRef<string>('')
 
-  const queryAndSetResults = useCallback(
-    async (text: string) => {
-      setLoading(RemoteDataState.Loading)
-      const result = await query(text)
-      setLoading(RemoteDataState.Done)
-      setResults(result)
-    },
-    [setLoading, setResults, query]
-  )
+  const queryAndSetResults = useCallback(async () => {
+    setLoading(RemoteDataState.Loading)
+    const result = await query(queryText.current)
+    setLoading(RemoteDataState.Done)
+    setResults(result)
+  }, [setLoading, setResults, query])
 
   useEffect(() => {
-    queryAndSetResults(data.query)
-  }, [data.query, queryAndSetResults])
+    // data.query gets updated often but doesn't actually change
+    // this limits querying to when the query actually changes
+    if (!queryText.current) {
+      queryText.current = data.query
+      queryAndSetResults()
+    }
+  }, [data, data.query, queryAndSetResults])
```

Comment:
removing all of these requirements so that this only runs on open, doesn't fix it?